### ISSUE TITLE
SAMPLES missing bug samples

### ISSUE THREAD

**Issue by ctsolomon** [MFEh2o/db#37]
The usual sampling protocol for benthic invertebrates in Long involves 4/4/2 samples at 1/3/8 m depth, in each basin. But for one sampling date (2017-06-05), there are 3/1/2 samples in EL and 3/3/2 samples in WL. I suspect that the samples were actually taken but had no bugs in them, and so no sampleID was created. We should check history of this and add the missing rows to SAMPLES table.

Note that Alex previously did a bunch of work trying to clean up this sort of issue in the bug table. All or most of this work is documented in Box, see "Database/rawFiles_csv/bugFixes_20171220/updateIDNotes_bugFixes.20171217.docx" and "Database/rawFiles_csv/fall2018_dbFixes/bugTableFix_20181018.r". I am surprised that those previous fixes didn't catch this issue; it may be worth talking to him to make sure that we should indeed add these rows to SAMPLES (i.e. that it's not the case that he left them out on purpose for some reason).

Note that the missing samples are, I think:
EL_E 1 m
EL_E 3 m
EL_N 3 m
EL_SE 3 m
WL_NW 1 m
WL_SW 3 m

**Comment by ctsolomon** [comment]
Another date with a similar problem: 2018-06-25, looks like these samples are missing:
WL_NW 3 m
WL_SW 3 m

**Comment by ctsolomon** [comment]
There are some similar issues with the Crazy Chiros samples. I have some code in "bug biomass 2019-07.r" that identifies the missing samples and re-creates them.

**Comment by ctsolomon** [comment]
Another issue to sort through, which came up in working with Shun to figure out what is going on with Crazy Chiro samples: in some cases he took samples but didn't process them. The samples still exist. So we presumably want some way for the database to show that these samples exist, but also to make clear that they weren't worked up (and so it's not like they are necessarily NFC).

**Comment by kaijagahm** [comment]
Decided in meeting on 9 June 2020 that we'll create a new table: BENTHIC_INVERT_SAMPLES, containing only the samples for inverts. This will be analogous to the FISH_SAMPLES table, and it will allow us to add another column, relevant only for the inverts, that indicates whether or not the sample was processed. This will allow us to add rows for the missing (unprocessed) samples.

Randi drew up a schema of which metadataID's should be pulled out into the BENTHIC_INVERT_SAMPLES table.
?	unknown
b	zoobenthos.Isotope.Sample.20110714
b	zoobenthos.Sample.20110714
b	BenthicInvert.Sample.20110714
x	BenthicPrimaryProduction.Sample.20110714
b	zoobenthos.Lipid.Sample.20120501
b	crazy.Chiro2015.2016
b	zoobenthos.Sample.20181114
x	Methane.Sample.20100929
x	DIsedcores.Samples.2010115
x	Methane.Sample.20110601
x	ZoopSurv.Sample.20110517
x	Limno.Sample.20110517
x	MethSurv.Sample.20120522
x	BenthicBacterialProduction.Analysis.20110616
x	Limno.Sample.20130416

**Comment by kaijagahm** [comment]
> There are some similar issues with the Crazy Chiros samples. I have some code in "bug biomass 2019-07.r" that identifies the missing samples and re-creates them

@ctsolomon I can't find this script in Box. Do you have it?

**Comment by kaijagahm** [comment]
Still to do on this issue:

1. Wait to hear back from Alex about whether there's a reason that some samples are still missing, or whether I should fill those in.
2. If the latter, fill in the missing samples (filling in BENTHIC_INVERT_SAMPLES this time, not SAMPLES!)
3. Figure out which Crazy Chiro samples Shun took but didn't process. Add those to BENTHIC_INVERT_SAMPLES, with "no" or "false" in the "processed" column.
4. Look through Chris's script "bug biomass 2019-07.r" (if I can get it) to see if anything further needs to be done about correcting missing rows for Crazy Chiro.

**Comment by kaijagahm** [comment]
1. Alex says that it's most likely that this is just a mistake, although it's a little strange that they would have missed it, since there was so much back and forth on the issue. Still, he suggests that I just move forward with adding those rows.

**Comment by kaijagahm** [comment]
Latest: have to figure out the sampling scheme in order to add the rows. Start by looking at Chris's script "bug biomass 2019-07.R", since he says he already did some of the work of figuring that out in this script.

**Comment by kaijagahm** [comment]
Ok, now I'm confused. @Randinotte labeled, above, the metadataID's that are related to bug samples, but they don't seem to line up with what I have in SAMPLES. I have the following: 

```
[1] "unknown"                                      "Methane.Sample.20100929"                     
 [3] "DIsedcores.Samples.2010115"                   "Clifford.20110310"                           
 [5] "Limno.Sample.20110517"                        "zoobenthos.Isotope.Sample.20110714"          
 [7] "zoobenthos.Sample.20110714"                   "ZoopSurv.Sample.20110517"                    
 [9] "Methane.Sample.20110601"                      "BenthicInvert.Sample.20110714"               
[11] "tPOCdep.Sample.20110617"                      "Piezometer.Sample.20140319"                  
[13] "BenthicPrimaryProduction.Sample.20110714"     "Staff.Gauge.Sample.20140319"                 
[15] "Limno.Sample.20120501"                        "ZoopCounts.20110601"                         
[17] "MethSurv.Sample.20120522"                     "zoop.horiz.tow.Sample.20120501"              
[19] "seston.point.Sample.20120501"                 "PeriphytonTile.Sample.20120601"              
[21] "zoobenthos.Lipid.Sample.20120501"             "zoop.tow.Sample.20120501"                    
[23] "MOLC.Sample.20120612"                         "BenthicBacterialProduction.Analysis.20110616"
[25] "Limno.Sample.20130416"                        "dvm.survey.05122013"                         
[27] "Limno.Sample.20140501"                        "GonadSurvey.20130509"                        
[29] "Piezometer.sensors.20190510"                  "Surveying.20190509"                          
[31] "crazy.Chiro2015.2016"                         "zoopCounts.20181114"                         
[33] "Limno.Sample.20160505"                        "zoobenthos.Sample.20181114"                  
[35] "GC6890.CH4.CO2.20120522"                      "zoopCounts.20191212"                         
[37] "Limno.Methane.Sample.20180416"                "Limno.pH.Sample.20190322"                    
[39] "fishScapesLimno.20190319"                     "Color.20110601"                              
[41] "DOC.20110601"      
```
Seems like there are several of these that weren't included in Randi's list that probably do relate to bugs.

**Comment by Randinotte** [comment]
I got that shorter list of metadataIDs by subsetting SAMPLES where SAMPLES$depthClass=="sediment" because all of the observations in BENTHIC_INVERTS have depthClass=="sediment".  Of the leftover ones from your comment above (copied here for clarity), none of them are related to benthic inverts:

**Longer list Kaija found**
[1] "Clifford.20110310"              "tPOCdep.Sample.20110617"        "Piezometer.Sample.20140319"    
 [4] "Staff.Gauge.Sample.20140319"    "Limno.Sample.20120501"          "ZoopCounts.20110601"           
 [7] "zoop.horiz.tow.Sample.20120501" "seston.point.Sample.20120501"   "PeriphytonTile.Sample.20120601"
[10] "zoop.tow.Sample.20120501"       "MOLC.Sample.20120612"           "dvm.survey.05122013"           
[13] "Limno.Sample.20140501"          "GonadSurvey.20130509"           "Piezometer.sensors.20190510"   
[16] "Surveying.20190509"             "zoopCounts.20181114"            "Limno.Sample.20160505"         
[19] "GC6890.CH4.CO2.20120522"        "zoopCounts.20191212"            "Limno.Methane.Sample.20180416" 
[22] "Limno.pH.Sample.20190322"       "fishScapesLimno.20190319"       "Color.20110601"                
[25] "DOC.20110601"   

**Ones I checked to make sure**
These are the ones wasn't familiar with, so I read their metadata to make sure they're not bug related. However, NONE of them were bug related. 
"Clifford.20110310", "tPOCdep.Sample.20110617", "MOLC.Sample.20120612", "dvm.survey.05122013" , "GonadSurvey.20130509",  "Surveying.20190509"

**Comment by ctsolomon** [comment]
Re samples that Shun took but didn't process: Kaija I will forward an email string to you that provides some answers.

**Comment by kaijagahm** [comment]
Re: Shun's samples, and the email string: here is the spreadsheet that I came up with after reading through that email string from Shun. 
[crazy chiro sampling structure.xlsx](https://github.com/MFEh2o/db/files/5022370/crazy.chiro.sampling.structure.xlsx)

Next to do: add the NFC and unprocessed rows to BENTHIC_INVERT_SAMPLES, and mark them as either NFC or unprocessed.

**Comment by kaijagahm** [comment]
Finding some weirdness as I add these Crazy Chiro samples.

2016-05-03 WL: should have W(3,4) and SE(3,4,5). Instead, has SW(3,4) and NW(3,4,5). Are those even valid sites in WL?

2016-05-18 WL: in addition to the samples it's supposed to have, there's also NW(3) and N(1). I'm not sure NW is a valid site. N is a valid site, but I don't think a depth of 1 makes sense? Yes, 1 is recorded for both depthTop and depthBottom.

**Comment by kaijagahm** [comment]
Okay, it appears that I need to re-make the spreadsheet based on what I'm finding in this current version of the database. Going to directly compare sites and samples between metadata and what we have in the database.

**Comment by kaijagahm** [comment]
@ctsolomon could you take a look at the new spreadsheet I've created to try to get to the bottom of these missing bug samples? I've looked through your email with Shun and attempted to identify which missing samples are NFC, which are unprocessed, and which are not accounted for. I did find some missing samples that weren't flagged as missing on the spreadsheet you sent Shun, as well as some mysterious extra samples, and some samples whose dates/sites may be typos. Would appreciate your input before I correct them.

The spreadsheet is here: https://docs.google.com/spreadsheets/d/1zDJBOMTbK8oL9EuNXdjb8ckPoOYxPQKMp6ZRuYBPKx8/edit?usp=sharing

**Comment by ctsolomon** [comment]
@kaijagahm I've now gone through that spreadsheet and left comments in all the places where you had queries for me. Should be possible to resolve most of them now, but pull me back in as needed.

**Comment by ctsolomon** [comment]
We should check that samples from the 2012 cross-lakes benthic invert survey are correctly recorded in the new BENTHIC_INVERT_SAMPLES table. They may already be correct - just occurs to me to check on this and make sure we don't have the NFC issue because I am looking at the "BenthicInvert.Sample.20110714.docx" metadata and seeing the table there of how many samples were taken on each date at each depth in these lakes. Regular bug samples in Long (EL and WL) served double-duty for that survey so those ones are probably all set already, but we should check on the other lakes.

**Comment by kaijagahm** [comment]
Good call, @ctsolomon. It's a little hard to check these because there's so much variation in the sampling scheme. Some of it is accounted for by that table in the metadata, but it looks like the samples recorded in the database deviate from that table a little bit.

Doing my best to account for those with common sense (e.g. if there were two 12m samples and two 14m samples, I could infer that those can all count towards 12m), I found the following discrepancies:

- RE should have four 1m samples on 2012-06-21, but it only has three. Possibly a missing NFC sample?
- CR only has three 12m samples on each of three dates, instead of the four that it's supposed to have (according to the metadata)
- BA is missing a 3m sample on 2012-05-22

Those are the only problems I could come up with. I'm a little skeptical that these are actually NFC, since there are so few of them. The CR issue could be a non-issue: it seems more likely that they decided to only take three samples and forgot to record it than that they happened to get one NFC sample at that depth (and only that depth) on each of the three dates. What do you think?

The RE and BA missing samples could be NFC, or they could have been mistakes where the samples were accidentally never taken. How should we go about figuring this out?

**Comment by ctsolomon** [comment]
I would bet these are real NFCs, in part because the paper that describes these data (Craig et al. 2015 Limnology and Oceanography) talks about 4 replicate samples at each sampled depth and doesn't mention deviations. Of course we might have decided that minor deviations weren't worth describing in the Methods; one way to check this would be to see if we can find Nikki's calculations. I'll share a folder with you from Nikki's archive - can you spend a few minutes to see if you can find the answer in there?

**Comment by kaijagahm** [comment]
Sure, happy to. Did you already share a link to that archive? I can't find it, but may have missed it if it was a while back in my email.

**Comment by kaijagahm** [comment]
From the document in Nikki's archive called "Record of bug samples taken for bug survey":

"B3 not taken in BW1" for lakeID == BA. This explains the missing 3m sample for BA on 2012-05-22 (BW1 is "bug week 1", and 2012-05-22 is indeed the first sampling date for BA. So that sample is actually missing, not NFC.

"B1 not taken in BW2" for lakeID == RE. This explains the missing 1m sample in RE. BW2 should be "bug week 2". It's not immediately obvious from my script (BENTHIC_INVERT_SAMPLES_create_gh37.R) whether the 2012-06-21 date is in BW2 for RE. It kind of looks like it's BW3, but that could either be a mistake or that first week (which looks different from the rest, anyway) could have been a trial week, since its right at the beginning of the season in May. In any case, seems reasonable to assume that that sample is actually missing, not NFC. 

The bug notes seem to show that only three 12m samples are supposed to be taken for CR: sites C, D, and E, but not A. If that's the case, then there's no problem with CR.

I feel comfortable concluding that none of these are NFC. I'm going to make a note in the metadata. I wonder if I should update the sampling scheme shown in the metadata doc to reflect what was actually taken. @ctsolomon do you think that's a good idea, or should that sampling scheme stay as is, as a record of what was supposed to happen? If the latter, then I think I should at least link to this "Record of bug samples" spreadsheet so we can recreate this search if need be.

**Comment by kaijagahm** [comment]
Heard back from Shun about missing Crazy Chiro samples. Able to move forward with most of them, or at least take our best informed guess.

**Comment by ctsolomon** [comment]
@kaijagahm, re whether to update the sampling scheme in the metadata doc: Let's have our cake and eat it too - let's have that doc indicate what the planned sampling strategy was and also indicate where samples were not taken.

**Comment by kaijagahm** [comment]
Sounds good. I added notes to this effect in the metadata doc. This issue is all set for a database update, which I'm going to put through this afternoon.

**Comment by kaijagahm** [comment]
Closed with db version 3.5.0, 9/30/2020

**Comment by kaijagahm** [comment]
I have a sinking feeling about this issue after finding this comment from Alex on 2018-11-20 (corresponding to the 
 tableBenthicInverts.20181114 updateID): 

> 6030 new entries from 2017 and 2018 were added to the database. CAUTION - numbers for 2018 seem to be somewhat low<U+0080>only ~2000 observations in 2018 compared to ~4000 in 2017. Things to consider; crew wasn't getting 'good' eckman dredges, pickers weren't doing a great job, or numbers are actually down.. hard to say. Another important caution -- Long lake samples follow 4/4/2 structure at 1,3,8m sites EXCEPT for dates 2017-06-05 and 2018-06-25, where for 2017-06-05... EL: at 1m, n=3 (missing the EL-E site), and at 3m n=1 (missing data from EL-E, N, SE). 8m sites are fine; where for 2018-06-25...samples from SW3 and NW3 were mislabelled so are missing data, not zerosWL: at 1m, n=3 (missing the WL-NW site), and at 3m n=3 (missing data from WL-SW). 8m sites are fine

Need to check on this. This comment indicates that some of the data are actually missing b/c mislabeled, not necessarily NBC.

**Comment by ctsolomon** [comment]
OK...sounds like we need to get back into it but it's potentially solvable. I was worried at first when I saw your note about a sinking feeling. Limnologists really don't like sinking.

Bring me in as needed as you sort through this.

**Comment by kaijagahm** [comment]
Steps to take here:
1. Check the current version of the database to see if what we have now matches what Alex wrote here.
2. If not, investigate discrepancies. Are there any rows that I added a while back that really should not have been added according to AR?
3. What about the photo archive?

**Comment by kaijagahm** [comment]
Okay, I checked the database and the photo archive.

To recap: 
> Long lake samples follow 4/4/2 structure at 1,3,8m sites EXCEPT for dates 2017-06-05 and 2018-06-25, where for 2017-06-05... EL: at 1m, n=3 (missing the EL-E site), and at 3m n=1 (missing data from EL-E, N, SE). 8m sites are fine; where for 2018-06-25...samples from SW3 and NW3 were mislabelled so are missing data, not zerosWL: at 1m, n=3 (missing the WL-NW site), and at 3m n=3 (missing data from WL-SW). 8m sites are fine

**2017-06-05**
The database matches what Alex wrote: no sample for EL_E_1 or for EL_E_3, EL_N_3, or EL_SE_3. But the photo archive doesn't match. Photo archive has no photo for EL_E_1 or EL_E_3, which matches what Alex said. But it *does* have a photo for EL_N_3 and EL_SE_3, and there's also an unlabeled image that has some bugs. 

**2018-06-25**
Alex says that WL_NW_1 is missing, but it's present in the database AND there's a photo for it.
Alex says that WL_SW_3 is missing. It is indeed missing from the database and photos.
Alex says that there are three WL_3m samples (rather than 4). But I see only two in the database: WL_S_3 and WL_N_3. The photo archive has photos for those two, as well as WL_NW_3, which is missing from the database.

@ctsolomon I wonder if it makes sense to lump these checks in with whatever checks we do for #88, since both are going to involve solving photo-related mysteries?

**Comment by ctsolomon** [comment]
Yes that seems fine. Let's talk tomorrow at meeting about how to get Amaryllis and me started on this.

**Comment by kaijagahm** [comment]
Sounds good. That's already on the agenda, so we'll talk bugs for sure.

**Comment by kaijagahm** [comment]
NTS: remember to include this issue when talking about #88.

**Comment by kaijagahm** [comment]
Resolved with #88.
